src/main/java/io/luna/game/model/item/shop/Shop.java

```diff
@@ -251,9 +252,41 @@ public boolean sell(Player player, int index, int sellAmount) {
      * @return The sell value.
      */
     private int computeSellValue(Item item, boolean total) {
-        int value = (int) Math.floor(item.getItemDef().getValue() / 2);
-        value = value <= 0 ? 1 : value;
-        return total ? value * item.getAmount() : value;
+        int value = 0;
+        int currentStock;
+        int defaultStock = 0;
+        int stockDiff = 0;
+
+        //Check if the shop already have this item in stock
+        if(container.contains(item)) {
+
+            //Find out how many of that item are currently in stock in the shop
+            currentStock = container.computeAmountForId(item.getId());
+
+            //Find out how many of that item are in stock by default in the shop
+            if(amountMap[container.computeIndexForId(item.getId()).getAsInt()].isPresent())
+            defaultStock = amountMap[container.computeIndexForId(item.getId()).getAsInt()].getAsInt();
```

Comment:
computeIndexForId does a lookup so you probably want to cache the value
```java
var index = container.computeIndexForId(item.getId());
int defaultStock = index.isPresent() ? amountMap[index.getAsInt()].orElse(0) : 0;
```
This is still a bit ugly, so ideally in the future amountMap should be changed to `int[]` rather than `OptionalInt[]`